internal/command/test.go

```diff
@@ -544,6 +577,16 @@ func (runner *TestFileRunner) ExecuteTestRun(ctx context.Context, run *moduletes
 		return state, false
 	}
 
+	evalCtx, evalDiags := getEvalContextForTest(runner.States, config, runner.Suite.GlobalVariables)
+	run.Diagnostics = run.Diagnostics.Append(evalDiags)
+
+	diags := transformTestFileProviderConfigs(file, evalCtx)
```

Comment:
I think this part of the code should be executed on a file level instead of run level. We would have the same results regardless of what run is being executed.